hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerV2Impl.java

```diff
@@ -299,8 +299,9 @@ public void openPipeline(PipelineID pipelineId) throws IOException {
       }
       if (pipeline.getPipelineState() == Pipeline.PipelineState.ALLOCATED) {
         LOG.info("Pipeline {} moved to OPEN state", pipeline);
-        stateManager.updatePipelineState(pipelineId.getProtobuf(),
-            HddsProtos.PipelineState.PIPELINE_OPEN);
+        HddsProtos.Pipeline pipelineProto = stateManager.updatePipelineState(
+            pipelineId.getProtobuf(), HddsProtos.PipelineState.PIPELINE_OPEN);
+        pipeline = Pipeline.getFromProtobuf(pipelineProto);
```

Comment:
We don't have to construct the `Pipeline` from proto object. `stateManager#getPipeline` should be sufficient to get the updated pipeline.

Reply:
Updated.